dataland-frontend/src/components/forms/CreateEUTaxonomyForNonFinancials.vue

```diff
@@ -142,21 +455,143 @@ export default defineComponent({
+     *
+     * @param dateValue new date value
+     */
+    updateFiscalYearEndHandler(dateValue: Date) {
+      this.convertedFiscalYearEnd = getHyphenatedDate(dateValue);
+      this.fiscalYearEnd = dateValue;
+    },
+
+    /**
+     * Updates the date of a single report file
+     *
+     * @param index file to update
+     * @param dateValue new date value
+     * @param whichSetOfFiles which set of files will be edited
+     */
+    updateReportDateHandler(index: number, dateValue: Date, whichSetOfFiles: WhichSetOfFiles) {
+      this.updatePropertyFilesUploaded(index, "convertedReportDate", dateValue, whichSetOfFiles);
+      this.updatePropertyFilesUploaded(index, "reportDate", getHyphenatedDate(dateValue), whichSetOfFiles);
+    },
+
+    /**
+     * Add files to object filesToUpload
+     *
+     * @param event full event object containing the files
+     * @param event.originalEvent event information
+     * @param event.files files
+     */
+    onSelectedFilesHandler(event: { files: Record<string, string>[]; originalEvent: Event }): void {
+      if (event.files.length) {
+        event.files.forEach((file) => {
+          if (this.uploadFiles.some((existingFile) => existingFile.name === file.name.split(".")[0])) {
+            file["nameAlreadyExists"] = "true";
+          } else {
+            file["convertedReportDate"] = file["convertedReportDate"] ?? "";
+            file["documentId"] = file["documentId"] ?? "";
+            file["reportDate"] = file["reportDate"] ?? "";
+          }
+        });
+        this.filesToUpload = [...event.files] as ExtendedFile[];
+      } else {
+        return;
+      }
+    },
+
+    /**
+     * Update property in uploaded files
+     *
+     * @param indexFileToUpload Index number of the report
+     * @param property Property which is to be updated
+     * @param value Value to which it is to be changed
+     * @param whichSetOfFiles which set of files will be edited
+     */
+    updatePropertyFilesUploaded(
+      indexFileToUpload: number,
+      property: string,
+      value: string | Date,
+      whichSetOfFiles: WhichSetOfFiles
+    ) {
+      if (Object.prototype.hasOwnProperty.call(this[whichSetOfFiles][indexFileToUpload], property)) {
+        this[whichSetOfFiles][indexFileToUpload][property] = value;
+        this[whichSetOfFiles] = [...this[whichSetOfFiles]];
+      }
+    },
+
+    /**
+     * Remove report from files uploaded
+     *
+     * @param fileToRemove File To Remove
+     * @param fileRemoveCallback Callback function removes report from the ones selected in formKit
+     * @param index Index number of the report
+     */
+    removeReportFromFilesToUpload(
+      fileToRemove: Record<string, string>,
+      fileRemoveCallback: (x: number) => void,
+      index: number
+    ) {
+      fileRemoveCallback(index);
+      this.filesToUpload = this.filesToUpload.filter((el) => {
+        return el.name !== fileToRemove.name;
+      });
```

Comment:
Those are exact duplicate from the CreateEuTaxoFinancials.  Can we avoid this?

Reply:
Yes.